generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/state/TypesCache.kt

```diff
@@ -64,6 +71,20 @@ internal class TypesCache(private val supportedPackages: List<String>) : Closeab
         return null
     }
 
+    private fun getCacheKey(type: KType, inputType: Boolean, annotations: List<Annotation> = emptyList()): TypesCacheKey {
+        val unionAnnotation = annotations.getUnionAnnotation()
+
+        return if (unionAnnotation != null) {
```

Comment:
This is handled already before when we call `kClass.isUnion`

Reply:
I don't see where that is happening. This is called by `get(type: KType, inputType: Boolean, annotations: List<Annotation>)`, which is called by `objectFromReflection`, which is called by any number of potential objects and properties.

Reply:
@HGF Are you sure? Because calling `objectFromReflection` run the util `kClass.isUnion` which checks for `Any`.

Can you provide a unit test that fails?

Reply:
If I run with this code, the schema gets generated as an `Int` not a union

```
class InvalidReturnType {
        @GraphQLUnion(name = "Number", possibleTypes = [Int::class])
        fun number1(): Int = 1
    }
```

Reply:
it's called in `generateGraphQLType` which is called after checking the cache, but the cache key will be wrong. I'll get a test written